app/controllers/api/databases.php

```diff
@@ -3302,9 +3303,31 @@ function updateAttribute(
                 }
             }
 
-            $relationships = \array_filter(
+            $relationships = array_filter(
                 $collection->getAttribute('attributes', []),
-                fn($attribute) => $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP
+                function (Document $attribute) use ($oldDocumentToBeUpdated, $newDocument) {
+                    if ($attribute->getAttribute('type') === Database::VAR_RELATIONSHIP) {
+                        $relationKey = $attribute->getAttribute('key');
+
+                        $oldRelationDocument = $oldDocumentToBeUpdated[$relationKey] ?? [];
+                        $newRelationDocuemntFromRequestData = $newDocument[$relationKey] ?? [];
+
+                        if (count($oldRelationDocument) !== count($newRelationDocuemntFromRequestData)) {
+                                // Return true if a difference is found in the relationships
+                                return true;
+                        }
+
+                        foreach ($oldRelationDocument as $key => $obj1) {
+                            $obj2 = $newRelationDocuemntFromRequestData[$key] ?? null;
+                            if (($obj1 instanceof Document && $obj2 instanceof Document && $obj1->getArrayCopy() !== $obj2->getArrayCopy())  || !($obj2 instanceof Document)) {
+                                // Return true if a difference is found in the relationships
+                                return true;
+                            }
+                        }
```

Comment:
I didn't get a chance to test this, but what if we have a setup like this:

```
A <-- two way, one-to-one --> B <-- two way, one-to-one --> C
```

Where the user has update permissions to A and C and they attempt to update A and C, but not B like:

```json
{
  "someString": "updated value",
  "b": {
    "$id": "asdf",
    "someString": "unchanged value",
    "c": {
      "$id": "zxcv",
      "someString": "updated value"
    }
  }
}
```

What would happen?

Reply:
Never mind. Let's not worry about this case. We should just reject this case because the user doesn't have access to B. However, this API call should succeed:

```
{
  "someString": "updated value",
  "b": "any_id"
}
```

Reply:
I think we should fix this case as well. It's perfectly fine for the user to update A and C through B right? 
I moved to a new logic as filtering out relationships is not a good way. 

I have changed my logic to skip Authorization if there is no changes to the document instead of filtering out relationships as I cannot go to C if B is filtered out.